app/recreate_phrases.py

```diff
@@ -0,0 +1,56 @@
+import os
+import logging
+
+import scipy.io.wavfile as wavfile
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def recreate_phrases(signal, sample_rate, samples_per_observation, predictions):
+    """
+    Creates WAV files for each distinct phrase in the prediction data.
+    """
+
+    phrases = []
+
+    # Seek along the predictions until we find a real speaker
+    current_speaker = 0
+    i = 0
+
+    while current_speaker <= 0:
+        current_speaker = predictions[i]
+        i += 1
+
+    # Continue along the current speaker until they change
+    start = i
+
+    while i < len(predictions):
+        # If speaker changed, this marks the end of the phrase
+        if current_speaker != predictions[i]:
+            end = i
+            phrases.append((start, end, current_speaker))
```

Comment:
Only matters if `SAMPLES_PER_OBSERVATION` is large, but I think `end = i - 1` here 🤔 On the other hand it doesn't hurt to pad a bit.